File: C#-Server/PromoItProject/PromoItProject.MicroService/UsersServices.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Newtonsoft.Json;
using PromoItProject.Entities;

namespace PromoItProject.MicroService
{
    public static class UsersServices
    {
        //http://localhost:7243/api/Users/Get
        //http://localhost:7243/api/Users/Get/id
        //http://localhost:7243/api/Users/Add
        //http://localhost:7243/api/Users/Update/id
        //http://localhost:7243/api/Users/Remove/id
        [FunctionName("Users")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "Users/{action}/{param?}")] HttpRequest req, string action, string param)
        {
            string cmdName = "Users." + action;
            try
            {
                ICommand command = MainManager.Instance.commandsManager.CommandList[cmdName];
                if (command != null)
                {
                    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
                    var result = command.Execute(param, requestBody);
                    if (result != null)
                    {
                        return new OkObjectResult(result);
                    }
                    else
                    {
                        MainManager.Instance.logger.LogError("An error occurred while executing the request");
                        return new BadRequestObjectResult("An error occurred while executing the request. Please try again in a few minutes, if the problem continues contact ProLobby.");
                    }
                }
                else
                {
                    MainManager.Instance.logger.LogError("Command not found");
                    return new NotFoundObjectResult("Command not found");
                }
            }
            catch (Exception ex)
            {
                MainManager.Instance.logger.LogException(ex.Message, ex);
                return new BadRequestObjectResult("An error occurred while executing the request. Please try again in a few minutes, if the problem continues contact ProLobby.");
            }

        }
    }
}

File: React-Client/promoit-client/src/services/users.services.js
import axios from "axios";

import { toast } from "react-toastify";
import { constUsersEndpoint } from "../constant/constantEndpoints";

//get all users
export const getUsersData = async () => {
  try {
    let endpoint = `${constUsersEndpoint}Get`;
    let response = await axios.get(endpoint);
    console.log(response);
    return response.data;
  } catch (error) {
    toast(
      "An error occurred while executing the request. Please try again in a few minutes, if the problem continues contact ProLobby."
    );
    console.error(error);
  }
};

//add user to DB
export const addUserToDb = async (user) => {
  try {
    console.log(user);
    let endpoint = `${constUsersEndpoint}Add`;
    await axios.post(endpoint, user);
  } catch (error) {
    toast(
      "An error occurred while executing the request. Please try again in a few minutes, if the problem continues contact ProLobby."
    );
    console.error(error);
  }
};
